pkg/component/runtime/command.go

```diff
@@ -460,21 +471,64 @@ func (c *CommandRuntime) getCommandSpec() *component.CommandSpec {
 	return nil
 }
 
-func attachOutErr(comp component.Component, cmdSpec *component.CommandSpec, typeStr string, binaryName string) process.CmdOption {
+func (c *CommandRuntime) syncLogLevels() {
+	ll, unitLevels := getLogLevels(c.current)
+	c.logStd.SetLevels(ll, unitLevels)
+	ll, unitLevels = getLogLevels(c.current) // don't want to share mapping of units (so new map is generated)
+	c.logErr.SetLevels(ll, unitLevels)
+}
+
+func attachOutErr(stdOut *logWriter, stdErr *logWriter) process.CmdOption {
 	return func(cmd *exec.Cmd) error {
-		dataset := fmt.Sprintf("elastic_agent.%s", strings.ReplaceAll(strings.ReplaceAll(binaryName, "-", "_"), "/", "_"))
-		logger := logger.NewWithoutConfig("").With("component", map[string]interface{}{
-			"id":      comp.ID,
-			"type":    typeStr,
-			"binary":  binaryName,
-			"dataset": dataset,
-		})
-		cmd.Stdout = newLogWriter(logger.Core(), cmdSpec.Log)
-		cmd.Stderr = newLogWriter(logger.Core(), cmdSpec.Log)
+		cmd.Stdout = stdOut
+		cmd.Stderr = stdErr
 		return nil
 	}
 }
 
+func createLogWriter(comp component.Component, cmdSpec *component.CommandSpec, typeStr string, binaryName string, ll zapcore.Level, unitLevels map[string]zapcore.Level, src logSource) *logWriter {
+	dataset := fmt.Sprintf("elastic_agent.%s", strings.ReplaceAll(strings.ReplaceAll(binaryName, "-", "_"), "/", "_"))
+	logger := logger.NewWithoutConfig("").With("component", map[string]interface{}{
+		"id":      comp.ID,
+		"type":    typeStr,
+		"binary":  binaryName,
+		"dataset": dataset,
+	})
+	return newLogWriter(logger.Core(), cmdSpec.Log, ll, unitLevels, src)
+}
+
+// getLogLevels returns the lowest log level and a mapping between each unit and its defined log level.
+func getLogLevels(comp component.Component) (zapcore.Level, map[string]zapcore.Level) {
+	baseLevel := zapcore.ErrorLevel
```

Comment:
[Blockerish]
`zapcore`? don't [pkg](https://github.com/elastic/elastic-agent/tree/main/pkg)/[core](https://github.com/elastic/elastic-agent/tree/main/pkg/core)/[logger](https://github.com/elastic/elastic-agent/tree/main/pkg/core/logger)/logger.go abstract it already? Now there are 2 different packages used to define the log level of `pkg/core/logger`. I'd be better to avoid this abstraction leak.

Reply:
The entire `logWriter` works on zapcore. The `logWriter` must work with `zapcore` because it need to bypass log level checking for `logp`.

Please see implementation of `logWriter` for information on why this needs to be done, or the PR where it was implemented.